internal/pkg/template/template.go

```diff
@@ -0,0 +1,101 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package template renders the static files under the "/templates/" directory.
+package template
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+
+	"github.com/aws/amazon-ecs-cli-v2/templates"
+	"github.com/gobuffalo/packd"
+)
+
+var box = templates.Box()
+
+// Parser is the interface that wraps the Parse method.
+type Parser interface {
+	Parse(path string, data interface{}, options ...ParseOption) (*Content, error)
+}
+
+// ReadParser is the interface that wraps the Read and Parse methods.
+type ReadParser interface {
```

Comment:
Shall we define those interfaces in the packages where they are used?

Reply:
I had it that way first but then I was defining the exact same interfaces in `cli/`, `deploy/cloudformation/stack/`, and `manifest/` so I thought at that point it was safe to define it in the producer package.